exercises/farewell-workflow/practice/translate.py

```diff
@@ -0,0 +1,34 @@
+import urllib.parse
+from temporalio import activity
+from temporalio.exceptions import ApplicationError
+
+
+class TranslateActivities:
+    def __init__(self, session):
+        self.session = session
+
+    @activity.defn
+    async def greet_in_spanish(self, name: str) -> str:
+        greeting = await self.call_service("get-spanish-greeting", name)
+        return greeting
+
+    # TODO: write an Activity method that calls the microservice to
+    # get a farewell message in Spanish. It will be identical to the
+    # method above, except the first argument to the callService
+    # method will be "get-spanish-farewell". You can name your
+    # method whatever you like.
+
+    # Utility method for making calls to the microservices
+    async def call_service(self, stem: str, name: str) -> str:
+        base = f"http://localhost:9999/{stem}"
+        url = f"{base}?name={urllib.parse.quote(name)}"
+
+        async with self.session.get(url) as response:
+            translation = await response.text()
+
+            if response.status >= 400:
+                raise ApplicationError(
+                    f"HTTP Error {response.status}: {translation}",
+                )
+
+            return translation
```

Comment:
```suggestion
            response.raise_for_status()
            return await response.text()
```

If you're not gonna reasonably differentiate retryable from non-retryable, you might as well just use the helper and make this code look really simple (and all other code where you do this). You don't really need the body or an `ApplicationError` for failures in your sample.